src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1575,6 +1589,62 @@ private static boolean withinRange(RepairSegment segment, Optional<RingRange> ra
     return !range.isPresent() || segmentIsWithinRange(segment, range.get());
   }
 
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions() {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions(String clusterName) {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .filter((subscription) -> clusterName.equals(subscription.getCluster()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public DiagEventSubscription getEventSubscription(UUID id) {
+    return createDiagEventSubscription(session.execute(getDiagnosticEventPrepStmt.bind(id)).one());
+  }
+
+  private static DiagEventSubscription createDiagEventSubscription(Row row) {
+    return new DiagEventSubscription(
+        Optional.of(row.getUUID("id")),
+        row.getString("cluster"),
+        Optional.of(row.getString("description")),
+        row.getList("include_nodes", String.class),
+        row.getList("events", String.class),
+        row.getBool("export_sse"),
+        row.getString("export_file_logger"),
+        row.getString("export_http_endpoint"));
+  }
+
+  @Override
+  public DiagEventSubscription addEventSubscription(DiagEventSubscription subscription) {
+    Preconditions.checkArgument(subscription.getId().isPresent());
+
+    session.execute(saveDiagnosticEventPrepStmt.bind(
+        subscription.getId().get(),
+        subscription.getCluster(),
+        subscription.getDescription(),
+        subscription.getIncludeNodes(),
+        subscription.getEvents(),
+        subscription.getExportSse(),
+        subscription.getExportFileLogger(),
+        subscription.getExportHttpEndpoint()));
+
+    return subscription;
+  }
+
+  @Override
+  public boolean deleteEventSubscription(UUID id) {
+    session.executeAsync(deleteDiagnosticEventPrepStmt.bind(id));
```

Comment:
We seem to have lost our conversation around this being async.
I think there's nothing worse than clicking a button that says "Delete" and nothing happens in the UI (and then I have to refresh so that it eventually disappears).
Using a synchronous query with `LOCAL_QUORUM` seems to me like a very reasonable tradeoff as this won't get called at fast pace. The performance gain with running it async is not worth having a UI that doesn't instantly reflects the change (I'm telling this because I've seen that inconsistent behavior in the UI when deleting subscriptions).

Are you ok with making a synchronous call here?

Reply:
See https://github.com/thelastpickle/cassandra-reaper/pull/441#discussion_r314040277

The UI should instantly reflect the change regardless, as it should simple deletes the html row (rather than refresh the page, or fetch the whole subscription list again and rebuild the table).

Also, the REST endpoint for the DELETE command returns `202 ACCEPTED`, so the client knows the delete is "being processed".